packages/core/src/blocks/ToggleWrapper/createToggleWrapper.ts

```diff
@@ -0,0 +1,191 @@
+      toggledState.set(editor.getBlock(block)!, false);
+
+      if (dom.contains(toggleAddBlockButton)) {
+        dom.removeChild(toggleAddBlockButton);
+      }
+    } else {
+      toggleWrapper.setAttribute("data-show-children", "true");
+      toggledState.set(editor.getBlock(block)!, true);
+
+      if (
+        editor.getBlock(block)?.children.length === 0 &&
+        !dom.contains(toggleAddBlockButton)
+      ) {
+        dom.appendChild(toggleAddBlockButton);
+      }
+    }
+  };
+  toggleButton.addEventListener("click", toggleButtonOnClick);
+
+  toggleWrapper.appendChild(toggleButton);
+  toggleWrapper.appendChild(renderedElement.dom);
+
+  const toggleAddBlockButton = document.createElement("button");
+  toggleAddBlockButton.className = "bn-toggle-add-block-button";
+  toggleAddBlockButton.textContent = "Empty toggle. Click to add a block.";
+  const toggleAddBlockButtonMouseDown = (event: MouseEvent) =>
+    event.preventDefault();
+  toggleAddBlockButton.addEventListener(
+    "mousedown",
+    toggleAddBlockButtonMouseDown,
+  );
+  const toggleAddBlockButtonOnClick = () => {
+    // Adds a single empty child block.
+    editor.transact(() => {
+      // dom.removeChild(toggleAddBlockButton);
+
+      const updatedBlock = editor.updateBlock(block, {
+        // Single empty block with default type.
+        children: [{}],
+      });
+      editor.setTextCursorPosition(updatedBlock.children[0].id, "end");
+      editor.focus();
+    });
+  };
+  toggleAddBlockButton.addEventListener("click", toggleAddBlockButtonOnClick);
+
+  dom.appendChild(toggleWrapper);
+
+  let childCount = block.children.length;
+  const onEditorChange = editor.onChange(() => {
+    const newChildCount = editor.getBlock(block)?.children.length ?? 0;
+
+    if (newChildCount > childCount) {
+      // If a child block is added while children are hidden, show children.
+      if (toggleWrapper.getAttribute("data-show-children") === "false") {
+        toggleWrapper.setAttribute("data-show-children", "true");
+        toggledState.set(editor.getBlock(block)!, true);
+      }
+
+      // Remove the "add block" button as we want to show child blocks and
+      // there is at least one child block.
+      if (dom.contains(toggleAddBlockButton)) {
+        dom.removeChild(toggleAddBlockButton);
+      }
+    } else if (newChildCount === 0 && newChildCount < childCount) {
+      // If the last child block is removed while children are shown, hide
+      // children.
+      if (toggleWrapper.getAttribute("data-show-children") === "true") {
+        toggleWrapper.setAttribute("data-show-children", "false");
+        toggledState.set(editor.getBlock(block)!, false);
+      }
+
+      // Remove the "add block" button as we want to hide child blocks,
+      // regardless of whether there are child blocks or not.
+      if (dom.contains(toggleAddBlockButton)) {
+        dom.removeChild(toggleAddBlockButton);
+      }
+    }
```

Comment:
Hm, not a fan of contents of the editor subscribing to the editor state.

Can this not be achieved with the `update` API: https://prosemirror.net/docs/ref/#view.NodeView.update

Maybe not, since it seems to not give contentDom changes?